org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/service/ServiceArgumentsProvider.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) OSGi Alliance (2020). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.osgi.test.junit5.service;
+
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.support.AnnotationConsumer;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.osgi.test.common.service.ServiceConfiguration;
+import org.osgi.test.junit5.context.BundleContextExtension;
+
+public class ServiceArgumentsProvider
+	implements ArgumentsProvider, AnnotationConsumer<ServiceSource> {
+
+	private ServiceSource			serviceSource;
+
+
+	@SuppressWarnings("resource")
+	@Override
+	public Stream<? extends Arguments> provideArguments(ExtensionContext context) throws Exception {
+
+		BundleContext bundleContext = BundleContextExtension.getBundleContext(context);
+
+
+
+		ServiceConfiguration<?> sc = context.getStore(ServiceExtension.NAMESPACE)
+			.getOrComputeIfAbsent(serviceSource,
+				(serviceSource) -> new ServiceConfiguration<>(serviceSource.serviceType(), serviceSource.filter(),
+					serviceSource.filterArguments(), 0, serviceSource.timeout()).init(bundleContext),
+				ServiceConfiguration.class);
+
+		sc.waitForService(2000);
+		Stream<Arguments> stream = sc.getServiceReferences()
+			.stream()
+			.filter(Objects::nonNull)
+			.map((sr) -> {
+
+				List<Object> list = new ArrayList<>();
+				Optional<AnnotatedElement> oElement = context.getElement();
+				if (oElement.isPresent()) {
+					if (oElement.get() instanceof Method) {
+						Method method = (Method) oElement.get();
+						for (Parameter param : method.getParameters()) {
+							if (param.getType()
+								.isAssignableFrom(serviceSource.serviceType())) {
+								list.add(bundleContext.getService(sr));
```

Comment:
You must get the service yourself from the BundleContext. Use the ServiceConfiguration. It wraps a ServiceTracker which is tracking all the services.